overlord/README.md

```diff
@@ -0,0 +1,80 @@
+Notes on state and changes
+===========================
+
+State is central to the consistency and integrity of any snap system. It’s maintained by snapd by managing the external on-disk state of snap installations and its own persistent working state of metadata, expectations, and in-progress operations.
+
+Working persistent state is implemented by `overlord/state.State` with a global lock, `State.Lock/Unlock`, to govern updates. If state is modified after acquiring the lock, it’s atomically updated to disk when the lock is released.
+
+State managers
+---------------
+State managers are used to manage both the working state and the on-disk snap state. They all implement the `overlord.StateManager` interface. Code-wise together with a few other auxiliary components they live in `overlord` and its subpackages. `overlord.Overlord` itself is responsible for wiring and coordination of all of these
+
+In broad terms, state managers have assigned responsibilities for different subsystems, and these are in mostly orthogonal areas. They then participate in the management and bookkeeping of the state via various mechanisms.
+
+During startup, and after construction, an optional `StartUp` method is invoked on each state manager. This is followed by the activation of an *ensure loop* which calls a state manager’s corresponding Ensure method at least once every 5 minutes.
+
+The *ensure loop* is intended to initiate any automatic state management and corresponding transitions, state repair, and any other consistency-maintaining operations.
+
+`state.Change`
+---------------
+A `state.Change` is a graph of `state.Task` structs and their inter-dependencies as edges. The purpose of both a `state.Change` and `state.Task` is identified by their kind (which should be an explanatory string value).
+
+Time-consuming and user-initiated operations, usually initiated from the API provided by the `daemon` package, should be realized using the `state.Change` functionality.
```

Comment:
```suggestion
Time-consuming and user-initiated operations, usually initiated from the API provided by the `daemon` package, should be realised using the `state.Change` functionality.
```
I should have seen this earlier, sorry!

Reply:
Nitpick^2: done/performed/run would sound more natural since we're talking about operations

Reply:
I switched to performed